diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4b33b85e9..d9a2b79ed 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -468,8 +468,7 @@ if (NOT IQTREE_FLAGS MATCHES "single")
         link_directories(${PROJECT_SOURCE_DIR}/liblinux_arm)
         LIST(APPEND STATIC_LIBS "${PROJECT_SOURCE_DIR}/liblinux_arm/libomp.a" "${PROJECT_SOURCE_DIR}/liblinux_arm/libz.a")
       else()
-        link_directories(${PROJECT_SOURCE_DIR}/lib)
-        LIST(APPEND STATIC_LIBS "${PROJECT_SOURCE_DIR}/lib/libomp.a" "${PROJECT_SOURCE_DIR}/lib/libz.a")
+        find_package(OpenMP)
       endif()
     elseif (WIN32)
         if (BINARY32)
