diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4b33b85e9..95896c1b6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -468,8 +468,24 @@ if (NOT IQTREE_FLAGS MATCHES "single")
         link_directories(${PROJECT_SOURCE_DIR}/liblinux_arm)
         LIST(APPEND STATIC_LIBS "${PROJECT_SOURCE_DIR}/liblinux_arm/libomp.a" "${PROJECT_SOURCE_DIR}/liblinux_arm/libz.a")
       else()
-        link_directories(${PROJECT_SOURCE_DIR}/lib)
-        LIST(APPEND STATIC_LIBS "${PROJECT_SOURCE_DIR}/lib/libomp.a" "${PROJECT_SOURCE_DIR}/lib/libz.a")
+      find_package(OpenMP)
+        if (OPENMP_FOUND)
+            # Add OpenMP flags to compiler options
+            set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
+            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
+            set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${OpenMP_EXE_LINKER_FLAGS}")
+                 # Ensure the OpenMP runtime (libgomp) is linked with the static library
+            target_link_libraries(iqtree2 OpenMP::OpenMP_CXX gomp)  # Link OpenMP runtime (libgomp)
+                 # If USE_CMAPLE is enabled, also link OpenMP runtime for iqtree2-aa
+            if (USE_CMAPLE STREQUAL "ON")
+                target_link_libraries(iqtree2-aa OpenMP::OpenMP_CXX gomp)  # Link OpenMP runtime (libgomp)
+            endif()
+                 message("Set OpenMP flags and linked OpenMP runtime (libgomp)")
+        else()
+            message(WARNING "OpenMP not found, skipping OpenMP setup.")
+        endif()
+        # link_directories(${PROJECT_SOURCE_DIR}/lib)
+        # LIST(APPEND STATIC_LIBS "${PROJECT_SOURCE_DIR}/lib/libomp.a" "${PROJECT_SOURCE_DIR}/lib/libz.a")
       endif()
     elseif (WIN32)
         if (BINARY32)
