diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4b33b85e9..8cc064a61 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -722,18 +722,18 @@ endif()
 #include_directories("${PROJECT_BINARY_DIR}/zlib-1.2.7")
 
 
-if (NOT IQTREE_FLAGS MATCHES "nozlib")
-    find_package(ZLIB)
-endif()
-
-if(ZLIB_FOUND)
-  message ("Using system zlib")
-  include_directories(${ZLIB_INCLUDE_DIRS})
-else(ZLIB_FOUND)
-  message ("Using own zlib-1.2.7")
-  include_directories("${PROJECT_BINARY_DIR}/zlib-1.2.7" "${PROJECT_SOURCE_DIR}/zlib-1.2.7")
-  add_subdirectory(zlib-1.2.7)
-endif(ZLIB_FOUND)
+# if (NOT IQTREE_FLAGS MATCHES "nozlib")
+#     find_package(ZLIB)
+# endif()
+
+# if(ZLIB_FOUND)
+#   message ("Using system zlib")
+#   include_directories(${ZLIB_INCLUDE_DIRS})
+# else(ZLIB_FOUND)
+message ("Using own zlib-1.2.7")
+include_directories("${PROJECT_BINARY_DIR}/zlib-1.2.7" "${PROJECT_SOURCE_DIR}/zlib-1.2.7")
+add_subdirectory(zlib-1.2.7)
+# endif(ZLIB_FOUND)
 
 ##################################################################
 # subdirectories containing necessary libraries for the build
